packages/dapp/src/components/Main.tsx

```diff
@@ -20,17 +20,24 @@ import { connect, DispatchProp } from "react-redux";
 
 class Main extends React.Component<DispatchProp<any> & RouteComponentProps<any>> {
   public async componentDidMount(): Promise<void> {
-    const civil = getCivil();
-    this.props.dispatch!(addUser(civil.userAccount));
+    getCivil(this.onAccountUpdated);
     await initializeParameterizer(this.props.dispatch!);
     await initializeGovernment(this.props.dispatch!);
     await initializeProposalsSubscriptions(this.props.dispatch!);
     await initializeGovernmentParamSubscription(this.props.dispatch!);
     await initializeSubscriptions(this.props.dispatch!);
+  }
+
+  public onAccountUpdated = async (): Promise<void> => {
+    const civil = getCivil();
+    const tcr = civil.tcrSingletonTrusted();
```

Comment:
This is more for my own edification, but sometimes I see this in the code:

```
const civil = getCivil();
const tcr = civil.tcrSingletonTrusted();
```

And sometimes I see this:

```
const tcr = getTCR();
```

Is one preferable to the other? Or does the method of getting `tcr` differ in different situations?

Reply:
no real difference, the 2nd one is just shorthand. I think we need to rethink slightly how we handle the civil instance, so I'm not too concerned about this since I think it's going to change anyways.